전민주/Week 5/HW/app/src/main/java/com/mingmingmon/myapplication/LoginActivity.kt

```diff
@@ -0,0 +1,87 @@
+package com.mingmingmon.myapplication
+
+
+import android.app.Activity
+import android.content.Intent
+import android.os.Bundle
+import android.widget.Toast
+import androidx.activity.result.ActivityResultLauncher
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.appcompat.app.AppCompatActivity
+import com.mingmingmon.myapplication.databinding.ActivityLoginBinding
+
+class LoginActivity : AppCompatActivity(){
+    private lateinit var binding: ActivityLoginBinding
+    private lateinit var activityResultLancher: ActivityResultLauncher<Intent>
+    var user: User? = null
+
+    override fun onCreate(savedInstanceState : Bundle?){
+        super.onCreate(savedInstanceState)
+        binding = ActivityLoginBinding.inflate((layoutInflater))
+        setContentView(binding.root)
+
+        getUserInfoFromSignUPActivity()
+        clickSignUpButton()
+        clickLoginButton()
+
+    }
+
+    private fun getUserInfoFromSignUPActivity(){
+        activityResultLancher = registerForActivityResult(ActivityResultContracts.StartActivityForResult()){
+                result ->
+            if(result.resultCode == Activity.RESULT_OK){
+                val data: Intent? = result.data
+                if (data != null) {
+                    user = data?.getParcelableExtra("USER", User::class.java)
+                    binding.activityLoginEtID.setText(user!!.id)
+                    binding.activityLoginEtPassword.setText(user!!.password)
+                }
+                else
+                    warningToast("회원정보가 비어있습니다. 확인해주세요.")
+            }
+        }
+    }
+
+    private fun clickSignUpButton(){
+        val signUpButton = binding.activityLoginBtnJoin.setOnClickListener{
+            activityResultLancher.launch(
+                Intent(this, SignUpActivity::class.java).apply {
+                    this
+                }
+            )
+        }
```

Comment:
val을 활용해서 새로 변수를 만들어 주지 않아도 잘 작동할 것 같네요!!
```suggestion
        binding.activityLoginBtnJoin.setOnClickListener{
            activityResultLancher.launch(
                Intent(this, SignUpActivity::class.java).apply {
                    this
                }
            )
        }
```